packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation.dart

```diff
@@ -57,3 +137,33 @@ class NSError {
   /// A string containing the localized description of the error.
   final String localizedDescription;
 }
+
+/// The root class of most Objective-C class hierarchies.
+class NSObject {
```

Comment:
How will implementation work for inheritance across the method channel? Will each object's Pigeon API have to implement this individually?

Reply:
No, it should be possible to have `class WKWebViewHostApiImpl extends NSObjectHostAPIImpl`.

It should also also be possible to pass a `class WKWebViewFlutterApiImpl extends NSObjectFlutterApiImpl` to `class WKWebViewImpl extends NSObjectImpl` when implementing the callbacks.

Reply:
Actually it looks like `HostApi` will look more like this

```dart
class A extends B {
  static AHostApiImpl api = AHostApiImpl();
  
  Future<void> aMethod() {
    return api.aMethod(this);
  }
}

class B {
  static BHostApiImpl api = BHostApiImpl();
  
  Future<void> bMethod() {
    return api.bMethod(this);
  }
}
```

or 

```dart
class A extends B {
  A({AHostApiImpl? aHostApi, BHostApiImpl? bHostApi})
      : aHostApi = aHostApi ?? AHostApiImpl(),
        super(bHostApi: bHostApi);

  final AHostApiImpl aHostApi;

  Future<void> aMethod() {
    return aHostApi.aMethod(this);
  }
}

class B {
  B({BHostApiImpl? bHostApi}) : bHostApi = bHostApi ?? BHostApiImpl();

  final BHostApiImpl bHostApi;

  Future<void> bMethod() {
    return bHostApi.bMethod(this);
  }
}
```

The `FlutterApi`s are protocols/interfaces, so they can use `class WKWebViewFlutterApiImpl extends NSObjectFlutterApiImpl implements WKWebViewFlutterApi`